base/set.jl

```diff
@@ -17,6 +17,37 @@ Set() = Set{Any}()
 Construct a [`Set`](@ref) of the values generated by the given iterable object, or an
 empty set. Should be used instead of [`BitSet`](@ref) for sparse integer sets, or
 for sets of arbitrary objects.
+
+# Examples
+```jldoctest
+julia> animals = Set{String}(["cat","fish","dog","bird"])
+Set(["fish", "cat", "bird", "dog"])
+```
+
+You can let Julia infer the type:
+```jldoctest
+julia> animals = Set(["cat","fish","dog","bird"])
+Set(["fish", "cat", "bird", "dog"])
+
+julia> typeof(animals)
+Set{String}
+```
+
+You can create an empty set using the [`Set`](@ref) constructor function and then fill it using `push!` function:
```

Comment:
No need to ref `Set` multiple times.

This sentence could avoid using "you" and be written something like

> A common way of incrementally creating a `Set` is by first creating an empty set with `Se()` and then add the elements using the `push!` function.